apps/src/blockTooltips/DropletFunctionTooltip.js

```diff
@@ -67,14 +68,15 @@ var DropletFunctionTooltip = function (appMsg, definition) {
   /** @type {Array.<parameterInfo>} */
   this.parameterInfos = [];
 
-  var paramId = 0;
-  var paramName;
-  while (!!(paramName = appMsg[this.parameterNameKey(paramId)] ||
-                          msg[this.parameterNameKey(paramId)])) {
+  for (var paramId = 0; ; paramId++) {
```

Comment:
I'm not sure it's much better, but I think semantically this is more like a while loop.
```javascript
var paramID = 0, outOfParams = false, paramName;
while (!outOfParams) {
  paramName = getLocalization(this.parameterNameKey(paramId));
  if (!paramName) {
    outOfParams = true;
  } else {
    // Create paramInfo and push into collection...
  }
  paramId++;
}
```
If you'd like to make this more readable you could isolate the 'walk until out of params' logic from the generation of the paramInfo (which could be done with `map` once you know how many you have) or somehow extract most of the inner loop to a function like `getParamInfoForIndex` that returns something falsy when you're out of params.